packages/server/utils/rateLimiters/StubRateLimiter.ts

```diff
@@ -0,0 +1,18 @@
+import {RateLimiter, RateLimitStats} from './RateLimiter'
+
+type StubRateLimiterMode = 'always-allow'
+
+export class StubRateLimiter implements RateLimiter {
```

Comment:
-1 overall, I'd like to avoid stubs if at all possible because whether a stub passes or fails doesn't tell us anything about the code that's in production. 

It looks like the spirit of this PR is to circumvent the rate limiter while testing.

Could we instead add a `if (NODE_ENV === 'testing')` conditional here & achieve the same result?
https://github.com/ParabolInc/parabol/blob/33c3478b8699f0a61e092b533c36e4e6ac03c704/packages/server/graphql/rateLimit.ts#L31-L35

Reply:
The downside to using the global `if (NODE_ENV === 'testing')` behavior is that we'd globally disable rate limits for all unit tests. There are use-cases where we'd want to test that the rate limiter is working correctly in a test.

My thought is that we'd default to the stub rate limiter in the test environment. Then, after introducing a server context (see https://github.com/ParabolInc/parabol/issues/6018), it would be pretty straightforward to pass a real rate limiter in a specific testing context where you want to test rate limiting.

> I'd like to avoid stubs if at all possible because whether a stub passes or fails doesn't tell us anything about the code that's in production

I think the same logic applies to disabling with an environment variable check. Any way you shuck it; we're short-circuiting production logic to be able to run unit tests without hitting the rate limit.

That said, I'm happy to go in whatever direction you prefer. @mattkrick let me know how you'd like to proceed.

Reply:
From our sync conversation, I'm going to go the environment variable route for now. I won't squash the existing logic in case we choose to come back to this in the future.